### PR TITLE
build: Warning message was not telling which target it links to.

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -1481,7 +1481,7 @@ def check_can_link_together(self, t: BuildTargetTypes) -> None:
         if not self.uses_rust() and links_with_rust_abi:
             raise InvalidArguments(f'Try to link Rust ABI library {t.name!r} with a non-Rust target {self.name!r}')
         if self.for_machine is not t.for_machine and (not links_with_rust_abi or t.rust_crate_type != 'proc-macro'):
-            msg = f'Tried to tied to mix a {t.for_machine} library ("{t.name}") with a {self.for_machine} target "{self.name}"'
+            msg = f'Tried to link {self.name!r} for {self.for_machine} with {t.name!r} for {t.for_machine}.'
```
Comment (dcbaker) [inline mesonbuild/build.py:1484]: the change from explicitly quoting to using `!r` is a logically separate change, it should either be in a separate commit or dropped

**Review by dcbaker — CHANGES_REQUESTED** [review]
The commit description isn't really accurate. the original message would say (mins the extra word) "tried to mix a build library "foo" with a host target "bar"", which is admittedly awkward English, but ford in fact say that "foo" was being mixed into "bar".

I'd suggest something like: "Clarify warning when mixing host and build targets"

**Comment by xclaesse** [comment]
This was an old commit that I rebased, but you modified this message a while ago, so it's not relevant anymore: https://github.com/mesonbuild/meson/commit/47acce8ec93ae73b5a95f3b21e35abb4b3c21f6a.
